ftw/testrunner.py

```diff
@@ -61,29 +61,35 @@ def run_stage(self, stage, logger_obj=None, http_ua=None):
         input, waits for output then compares expected vs actual output
         http_ua can be passed in to persist cookies
         """
-        if not http_ua:
-            http_ua = http.HttpUA()
-        if stage.output.log_contains_str and logger_obj is not None:
-            start = datetime.datetime.now()
-            http_ua.send_request(stage.input)
-            end = datetime.datetime.now()
-            logger_obj.set_times(start, end)
-            lines = logger_obj.get_logs()           
-            self.test_log(lines, stage.output.log_contains_str, False)
-        # This could be condensed but its more understandable this way
-        elif stage.output.no_log_contains_str and logger_obj is not None:
+       
+        # Send our request (exceptions caught as needed)
+        if stage.output.expect_error:
+            try:
+                if not http_ua:
+                    http_ua = http.HttpUA()
+                start = datetime.datetime.now()
+                http_ua.send_request(stage.input)
+                end = datetime.datetime.now()
+            except errors.TestError as err:
```

Comment:
Why pass here? Are you saying if we get an error, then continue (because we expected it?). If so, i think we should use `with pytest.raises(errors.TestError) as err:`, with meaningful output that we caught it